src/components/Nav.tsx

```diff
@@ -11,6 +13,30 @@ import { Link } from "react-router-dom";
 import history from "../lib/history";
 import { userTypeStudent } from "../constants/general";
 
+// Feature flag imports
+import { useSelector } from "react-redux";
+import { RootState } from "../reducers/index";
+
+interface FeatureFlagElementProps {
+  element: React.ReactElement;
+  flag: keyof RootState["featureFlagState"];
+}
+
+const FeatureFlagElement: React.FC<FeatureFlagElementProps> = ({
+  element,
+  flag,
+}) => {
+  const enabled = useSelector(
+    (state: RootState) => state.featureFlagState[flag]
+  );
```

Comment:
You should use these hooks and they are automatically typed

https://github.com/WilliamsStudentsOnline/wso-react/blob/d3e763402a11a6e41a81b3737741c3df6077eb7f/src/lib/store.ts#L74-L76